flow/project.py

```diff
@@ -803,18 +950,28 @@ def isdisjoint(self, group):
         """
         return set(self).isdisjoint(set(group))
 
-    def _generate_id(self, job, operation_name=None, index=0):
+    def get_aggregate_hash(self, jobs):
+        return calc_id('-'.join(map(str, jobs)))
+
+    def job_ids_seperated(self, jobs, upto=None):
+        return '-'.join(map(str, jobs))
+
+    def _generate_id(self, jobs, operation_name=None, index=0):
         "Return an id, which identifies this group with respect to this job."
-        project = job._project
+        project = jobs[0]._project
+
+        for job in jobs:
+            assert project == job._project
```

Comment:
We should probably do this before we index into jobs.

Reply:
okay!

Reply:
Still not addressed.